File: Leetcode Problems/344.py
from typing import List

class Solution:
    def reverseString(self, s: List[str]) -> None:
        n = len(s)
        i = 0
        j = n - 1

        while i < j:
            s[i], s[j] = s[j], s[i]  
            i += 1
            j -= 1

if __name__ == '__main__':
    s = Solution()
    lst = ["h", "e", "l", "l", "o"]
    s.reverseString(lst)
    print(lst) 

File: Leetcode Problems/344.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        int n = s.size();
        int i=0;
        int j = n-1;

        while(i < j) {
            swap(s[i++], s[j--]);
        }
    }
};

int main() {
    Solution obj344;
    vector<char> s = {'h', 'e', 'l', 'l', 'o'};
    obj344.reverseString(s);
    for(auto i: s) {
        cout << i << " ";
    }
    return 0;
}